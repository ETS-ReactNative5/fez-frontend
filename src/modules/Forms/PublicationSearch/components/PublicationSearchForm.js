import React, {Component} from 'react';
import {Card, CardHeader, CardText} from 'material-ui/Card';
import {Field} from 'redux-form/immutable';
import PropTypes from 'prop-types';
import {HelpIcon, TextField} from 'uqlibrary-react-toolbox';
import RaisedButton from 'material-ui/RaisedButton';
import {isDOIValue, isPubMedValue} from '../validator';
import {locale} from 'config';

const JOURNAL_ARTICLE = 179;

export default class PublicationSearchForm extends Component {

    static propTypes = {
        title: PropTypes.string.isRequired,
        explanationText: PropTypes.string.isRequired,
        pristine: PropTypes.bool,
        handleSubmit: PropTypes.func,
        loadDoiResultsList: PropTypes.func,
        loadPubmedResultsList: PropTypes.func,
        loadTitleResultsList: PropTypes.func,
        formValues: PropTypes.object,
        help: PropTypes.object,
        searchTitleField: PropTypes.string
    };

    constructor(props) {
        super(props);
    }

    performSearch = (event) => {
        // TODO: fix form submit, all data fetching should be done outside of the form
        // workaround: if user clicks Enter on search field OR search button, search is initiated and form is submitted
        if (event && event.key && event.key !== 'Enter') {
            return;
        }

        const { formValues } = this.props;
        const fieldValue = formValues.get('doiSearch');

        if (isDOIValue(fieldValue)) {
            this.props.loadDoiResultsList(fieldValue);
        } else if (isPubMedValue(fieldValue)) {
            this.props.loadPubmedResultsList(fieldValue);
        } else {
            // Pass fieldValue map through to the form component
            this.props.loadTitleResultsList(JOURNAL_ARTICLE, fieldValue);
        }
    };

    render() {
        const {pristine, handleSubmit, title, help, explanationText} = this.props;
        const searchForPublicationInformation = locale.pages.addRecord.searchForPublication;

        return (
            <form ref="publicationSearchForm" onSubmit={handleSubmit}>
                <Card className="layout-card">
                    <CardHeader className="card-header">
                        <div className="columns is-gapless is-mobile">
                            <div className="column">
                                <h2 className="title is-4">{title}</h2>
                            </div>
                            <div className="column is-narrow is-helpicon">
                                {help && (
                                    <HelpIcon
                                        title={help.title}
                                        text={help.text}
                                        buttonLabel={help.buttonLabel}
                                    />
                                )}
                            </div>
                        </div>
                    </CardHeader>
                    <CardText className="body-1">
                        <div>{explanationText}</div>
                        <Field component={TextField}
                               name="doiSearch"
                               fullWidth
                               floatingLabelText={searchForPublicationInformation.defaultSearchFieldLabel}
                               autoComplete="off"
                               autoFocus
                               onKeyPress={this.performSearch}
                        />
                        <div style={{textAlign: 'right', marginTop: '20px'}}>
                            <RaisedButton
                                label={searchForPublicationInformation.defaultButtonLabel}
                                secondary
                                onTouchTap={this.performSearch}
                                disabled={pristine}
                                type="submit"
                            />
                        </div>
                    </CardText>
                </Card>
            </form>
        );
    }
}
