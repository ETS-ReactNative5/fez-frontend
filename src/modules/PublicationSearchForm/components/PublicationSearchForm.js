import React, {Component} from 'react';
import {Field} from 'redux-form/immutable';
import PropTypes from 'prop-types';
import {propTypes} from 'redux-form/immutable';
import {TextField, StandardCard} from 'uqlibrary-react-toolbox';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import {validation} from 'config';

export default class PublicationSearchForm extends Component {
    static propTypes = {
        ...propTypes, // all redux-form props
        onSkipSearch: PropTypes.func,
        locale: PropTypes.object
    };

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <StandardCard title={this.props.locale.title} help={this.props.locale.help}>
                <div>{this.props.locale.text}</div>
                <form onSubmit={this.props.handleSubmit}>
                    <div className="columns">
                        <div className="column">
                            <Field
                                component={TextField}
                                className="requiredField"
                                name="searchQuery"
                                fullWidth
                                floatingLabelText={this.props.locale.fieldLabels.search}
                                autoComplete="off"
                                autoFocus
                                validate={[validation.required]}/>
                        </div>
                        <div className="column is-narrow">
                            <RaisedButton
                                className="is-mui-spacing-button"
                                label={this.props.locale.submit}
                                fullWidth
                                secondary
                                onTouchTap={this.props.handleSubmit}
                                disabled={this.props.invalid}
                            />
                        </div>
                        {
                            this.props.onSkipSearch &&
                            <div className="column is-narrow">
                                <FlatButton
                                    className="is-mui-spacing-button"
                                    label={this.props.locale.skip}
                                    fullWidth
                                    onTouchTap={this.props.onSkipSearch}
                                />
                            </div>
                        }
                    </div>
                </form>
            </StandardCard>
        );
    }
}
