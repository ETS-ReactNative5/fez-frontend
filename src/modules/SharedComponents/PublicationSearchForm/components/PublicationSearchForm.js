import React, {Component} from 'react';
import {Field} from 'redux-form/immutable';
import PropTypes from 'prop-types';
import {propTypes} from 'redux-form/immutable';

import {StandardCard} from 'modules/SharedComponents/Toolbox/StandardCard';
import {TextField} from 'modules/SharedComponents/Toolbox/TextField';

import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import {validation} from 'config';

import Grid from '@material-ui/core/Grid';

export default class PublicationSearchForm extends Component {
    static propTypes = {
        ...propTypes, // all redux-form props
        onSkipSearch: PropTypes.func,
        locale: PropTypes.object
    };

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <StandardCard title={this.props.locale.title} help={this.props.locale.help}>
                <Grid container>
                    <Grid item>
                        {this.props.locale.text}
                    </Grid>
                </Grid>
                <form onSubmit={this.props.handleSubmit}>
                    <Grid container spacing={8} alignItems={'flex-start'}>
                        <Grid item style={{flexGrow: 1}}>
                            <Field
                                component={TextField}
                                required
                                name="searchQuery"
                                fullWidth
                                label={this.props.locale.fieldLabels.search}
                                autoFocus
                                validate={[validation.required]}/>
                        </Grid>
                        <Grid item>
                            <RaisedButton
                                style={{marginTop: 22}}
                                label={this.props.locale.submit}
                                fullWidth
                                secondary
                                onClick={this.props.handleSubmit}
                                disabled={this.props.invalid}
                            />
                        </Grid>
                        {
                            this.props.onSkipSearch &&
                            <Grid item>
                                <FlatButton
                                    style={{marginTop: 22}}
                                    label={this.props.locale.skip}
                                    fullWidth
                                    onClick={this.props.onSkipSearch}
                                />
                            </Grid>
                        }
                    </Grid>
                </form>
            </StandardCard>
        );
    }
}
