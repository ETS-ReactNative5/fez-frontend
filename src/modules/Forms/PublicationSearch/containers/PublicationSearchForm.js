import {connect} from 'react-redux';

import {reduxForm, getFormValues} from 'redux-form/immutable';
import PublicationSearchForm from '../components/PublicationSearchForm';
import {validate} from '../validator';
import Immutable from 'immutable';
import {doiSearch, pubMedSearch, titleSearch} from '../actions';

let PublicationSearchFormContainer = reduxForm({
    form: 'PublicationSearchForm',
    validate
})(PublicationSearchForm);

PublicationSearchFormContainer = connect((state) => {
    return {
        formValues: getFormValues('PublicationSearchForm')(state) || Immutable.Map({})
    };
}, dispatch => {
    return {
        doiSearch: doi => dispatch(doiSearch(doi)),
        pubMedSearch: pubMedId => dispatch(pubMedSearch(pubMedId)),
        titleSearch: (rekDisplayType, title) => dispatch(titleSearch(rekDisplayType, title))
    };
})(PublicationSearchFormContainer);

export default PublicationSearchFormContainer;
