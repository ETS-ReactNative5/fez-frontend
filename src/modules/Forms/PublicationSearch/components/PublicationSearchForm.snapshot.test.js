jest.dontMock('./PublicationSearchForm');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import PublicationSearchForm from './PublicationSearchForm';

function setup() {
    const props = {
        title: 'Do Something',
        explanationText: 'This is some text that explains what this form block displays',
        defaultSearchFieldLabel: 'Search field label',
        defaultButtonLabel: 'Button Label',
        helpText: 'Some help text content',
        helpTitle: 'Help Title'
    };
    return shallow(<PublicationSearchForm {...props} />);
}

describe('PublicationSearchForm page snapshots tests', () => {
    it('renders default about page', () => {
        const wrapper = setup();
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
