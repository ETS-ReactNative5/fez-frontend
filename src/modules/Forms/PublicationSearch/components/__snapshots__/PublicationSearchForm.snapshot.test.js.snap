// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publication search form snapshots tests renders default publication search form page 1`] = `
<form>
  <Card
    className="layout-card"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardHeader
      avatar={null}
      className="card-header"
    >
      <div
        className="columns is-gapless is-mobile"
      >
        <div
          className="column"
        >
          <h2
            className="title is-4"
          >
            Component Title
          </h2>
        </div>
        <div
          className="column is-narrow is-helpicon"
        />
      </div>
    </CardHeader>
    <CardText
      className="body-1"
    >
      <div>
        Component Search Text
      </div>
      <Field
        autoComplete="off"
        autoFocus={true}
        component={[Function]}
        floatingLabelText="Enter DOI, Pubmed Id or Title"
        fullWidth={true}
        name="doiSearch"
        onKeyPress={[Function]}
      />
      <div
        style={
          Object {
            "marginTop": "20px",
            "textAlign": "right",
          }
        }
      >
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Search"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={false}
          secondary={true}
          type="submit"
        />
      </div>
    </CardText>
  </Card>
</form>
`;
